src/atlas/mesh/BuildMeshFromConnectivities.h

```diff
@@ -0,0 +1,52 @@
+/*
+ * (C) Copyright 2023 UCAR.
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+#pragma once
+
+#include "atlas/mesh/Mesh.h"
+
+#include <array>
+#include <vector>
+
+namespace atlas {
+
+//-----------------------------------------------------------------------------
+
+/**
+ * \brief Construct a Mesh by importing external connectivity data
+ *
+ * Given a list of nodes and corresponding cell-to-node connectivity, sets up a Mesh. Not all Mesh
+ * fields are initialized, but enough are to build halos and construct a NodeColumns FunctionSpace.
+ *
+ * The inputs lons, lats, ghosts, global_indices, remote_indices, and partitions are vectors ranging
+ * over the nodes locally owned by (or in the ghost nodes of) the MPI task. The global index is a
+ * uniform labeling of the nodes across all MPI tasks; the remote index is the 0-based vector index
+ * for the node on its owning task.
+ *
+ * The tri/quad connectivities (boundary_nodes and global_indices) are vectors ranging over the
+ * cells owned by the MPI task. Each cell is defined by a list of nodes defining its boundary;
+ * note that each boundary node must be locally known (whether as an owned of ghost node on the
+ * MPI task), in other words, must be an element of the node global_indices. The cell global index
+ * is, here also, a uniform labeling over the of the cells across all MPI tasks.
+ *
+ * Some limitations of the current implementation (but not inherent):
+ * - can only set up triangle and quad cells.
+ * - cannot import halos, i.e., cells owned by other MPI tasks; halos can still be subsequently
+ *   computed by calling the BuildMesh action.
```

Comment:
@ytremolet the current design of this PR means we'll ask the JEDI models to call the `build_halo` action after creating the mesh but before constructing the functionspace. It's one very simple line of code, so not a big deal. 

We could streamline a little either by...
1. adding a bool option here to call `build_halo` internally to this function
2. expanding the interface so models can import their halos directly
 
Personally I like the current design, because it keeps the responsibilities clear and makes the halo-building step easy to spot. Do you have a different opinion as to whether either of these options is significantly better than the current state?

Reply:
It seems fine. The test doesn't cover that part (or I missed it). That's ok because it's not part of the new function, but do you have an example (in fv3 or other?) to get a better idea what it looks like from the user point of view?

Reply:
@ytremolet I don't have something very clean, but on fv3-jedi branch `feature/fmahebert_test_atlas_mesh_builder` you can see (note to others, this is not a public repo) a code block in Geometry.cc that sets up the mesh, builds a halo, and tests some halo exchange; in the fortran you can see how the connectivity is inferred from the grid structure; all the above is mixed in which other small changes I was testing but didn't take the time to clean out (sorry).

Reply:
I had a look at your example. It looks reasonable given the information we want to get from the models. And for now the halo is not shocking, this is code that is only needed once for each model, it won't be repeated all over the place so I would be happy with the current implementation.